agent/src/com/cloud/agent/Agent.java

```diff
@@ -412,7 +412,8 @@ protected void reconnect(final Link link) {
             try {
                 _connection.start();
             } catch (final NioConnectionException e) {
-                throw new CloudRuntimeException("Unable to start the connection!", e);
+               s_logger.info("Attempted to connect to the server, but received an unexpected exception, trying again...");
```

Comment:
1. I think you should fix line 230/238, that's the root cause of the issue. line 415 is not.
2. line 416 can be removed.